src/components/Cart/Cart.js

```diff
@@ -0,0 +1,73 @@
+import React, { useEffect } from 'react';
+
+import { useShopify } from '../../data/reducers/index';
+import LineItem from './LineItem';
+
+function Cart() {
+    const {
+        cartStatus,
+        closeCart,
+        openCart,
+        checkoutState,
+        setCount,
+    } = useShopify();
+
+    function handleOpen( event ) {
+        event.preventDefault();
+        openCart();
+    }
+
+    function handleClose( event ) {
+        event.preventDefault();
+        closeCart();
+    }
+
+    function openCheckout( event ) {
+        event.preventDefault();
+        // window.open(checkoutState.webUrl) // opens checkout in a new window
+		window.location.replace(checkoutState.webUrl) // opens checkout in same window
+    }
+
+    useEffect(() => {
+        const button = document.querySelector( "button.App_view-cart" );
+
+        if ( cartStatus === true ) {
+            button.classList.add( "hide" );
+        } else {
+            button.classList.remove( "hide" );
+        }
+
+        function getCount() {
+            let lineItems =
+                checkoutState.lineItems && checkoutState.lineItems.length > 0
+                ? checkoutState.lineItems
```

Comment:
Is there ever a case where checkoutState.lineItems.length > 0 but checkoutState.lineItems object is NULL?

Reply:
> Is there ever a case where checkoutState.lineItems.length > 0 but checkoutState.lineItems object is NULL?

I don't think so but the example that Shopify had checked for both cases

Reply:
I see, all good in that case.